eam/object/object_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package object_test
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/eam"
+	"github.com/vmware/govmomi/eam/object"
+	"github.com/vmware/govmomi/eam/simulator"
+	vcsim "github.com/vmware/govmomi/simulator"
+)
+
+var (
+	client struct {
+		*eam.Client
+		ctx context.Context
+		eam object.EsxAgentManager
+		vim *govmomi.Client
+	}
+)
+
+func TestMain(m *testing.M) {
+	client.ctx = context.Background()
+
+	// Define a new model for vC Sim.
```

Comment:
If you haven't seen it, `simulator.Test` can simplify vcsim startup/teardown: https://github.com/vmware/govmomi/blob/06a88b327a527a3ae3e009b459dae3ee18bdd8e0/object/common_test.go#L48

And eam will be auto-registered via `RegisterEndpoint` as long as there's an import of eam/simulator